src/ResourceManager/KeyVault/Commands.KeyVault/Commands/ManagedStorageAccounts/RemoveAzureKeyVaultManagedStorageAccount.cs

```diff
@@ -0,0 +1,78 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.KeyVault.Models;
+using System.Globalization;
+using System.Management.Automation;
+using KeyVaultProperties = Microsoft.Azure.Commands.KeyVault.Properties;
+
+namespace Microsoft.Azure.Commands.KeyVault
+{
+    [Cmdlet( VerbsCommon.Remove, CmdletNoun.AzureKeyVaultManagedStorageAccount,
+        SupportsShouldProcess = true,
+         ConfirmImpact = ConfirmImpact.High,
+        HelpUri = Constants.KeyVaultHelpUri )]
+    [OutputType( typeof( ManagedStorageAccount ) )]
+    public class RemoveAzureKeyVaultManagedStorageAccount : KeyVaultCmdletBase
+    {
+        #region Input Parameter Definitions
+        [Parameter( Mandatory = true,
+            Position = 0,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Vault name. Cmdlet constructs the FQDN of a vault based on the name and currently selected environment." )]
+        [ValidateNotNullOrEmpty]
+        public string VaultName { get; set; }
+
+        [Parameter( Mandatory = true,
+            Position = 1,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Key Vault managed storage account name. Cmdlet constructs the FQDN of a managed storage account name from vault name, currently " +
+                          "selected environment and manged storage account name." )]
+        [ValidateNotNullOrEmpty]
+        [Alias( Constants.StorageAccountName )]
+        public string AccountName { get; set; }
```

Comment:
@kunalcha this is a pattern that we are trying to enforce across the board now for new cmdlets. It allows generic cmdlets to return objects with properties such as `Name`, `ResourceGroupName`, etc. and have those piped to this cmdlet by property name. I don't think it will cause confusion with the different cmdlets in the set since is an alias, so it will not show up when the user is tabbing through the possible parameters and in the help (`Name` will be mentioned in the properties part of the parameter as an alias). Let me know if you have any other questions.